src/userSync.js

```diff
@@ -0,0 +1,132 @@
+import * as utils from 'src/utils';
+
+const userSync = exports;
+// A queue of user syncs for each adapter
+const queue = {
+  image: []
+};
+
+/**
+ * @function fireSyncs
+ * @summary Trigger all user syncs in the queue
+ * @private
+ */
+function fireSyncs() {
+  let bodyElem = document.getElementsByTagName('body')[0];
+  try {
+    // Fire image pixels
+    queue.image.forEach((sync) => {
+      let bidderName = sync[0];
+      let trackingPixelUrl = sync[1];
+      let removeOnLoad = true;
+      utils.logMessage(`Invoking image pixel user sync for bidder: ${bidderName}`);
+      // insertAdjacentHTML expects HTML string - convert DOM object to string
+      let img = userSync.createImgObject(trackingPixelUrl, removeOnLoad);
+      if (img) {
+        bodyElem.insertAdjacentHTML('beforeend', img.outerHTML);
+      }
+    });
+    // Reset the image pixel queue
+    queue.image = [];
+  }
+  catch (e) {
+    utils.logError('Error firing user syncs', e);
+  }
+}
+
+/**
+ * @function hideNode
+ * @summary Modifies a DOM element to be hidden from user sight
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function hideNode(elementNode) {
+  elementNode = elementNode.cloneNode();
+  elementNode.height = 0;
+  elementNode.width = 0;
+  elementNode.style.display = 'none';
+  return elementNode;
+}
+
+/**
+ * @function setIdToNode
+ * @summary Adds a unique ID to a DOM element
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function setIdToNode(elementNode) {
+  elementNode = elementNode.cloneNode();
+  elementNode.id = utils.getUniqueIdentifierStr();
+  return elementNode;
+}
+
+/**
+ * @function hideAndIdElem
+ * @summary Functionally compose a DOM element to be hidden and add an ID
+ * @private
+ * @params {object} elementNode A valid DOM element
+ * @returns {object} A valid DOM element
+ */
+function hideAndIdElem(elementNode) {
+  return hideNode(setIdToNode(elementNode));
+}
+
+/**
+ * @function createImgObject
+ * @summary Create an img DOM element for sending a pixel. Made public for test purposes
+ * @public
+ * @params {string} url The URL for the image pixel
+ * @params {boolean} removeOnLoad Remove this img element once the endpoint is reached
+ * @returns {object} A valid DOM element
+ */
+userSync.createImgObject = function(url, removeOnLoad) {
+  if (!url) {
+    return;
+  }
+  const img = hideAndIdElem(new Image());
```

Comment:
this function seems a bit... odd. Is there reason to have all these smaller functions so you can compose other things in the future?